File: backend/app/models/platform.rb
# == Schema Information
#
# Table name: platforms
#
#  id            :bigint           not null, primary key
#  is_system     :boolean          default(FALSE), not null
#  name          :string(100)      not null
#  platform_type :string           not null
#  short_name    :string(20)       not null
#  created_at    :datetime         not null
#  updated_at    :datetime         not null
#
# Indexes
#
#  index_platforms_on_short_name  (short_name) UNIQUE
#
class Platform < ApplicationRecord
  has_many :platform_accounts, dependent: :restrict_with_error

  enum :platform_type, {
    broker:      "broker",
    mf_platform: "mf_platform",
    direct:      "direct",
    other:       "other"
  }, validate: true

  validates :name,       presence: true
  validates :short_name, presence: true, uniqueness: true
end

File: backend/app/models/platform.py
import enum
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Enum, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class PlatformType(str, enum.Enum):
    broker = "broker"
    mf_platform = "mf_platform"
    direct = "direct"
    other = "other"


class Platform(Base):
    __tablename__ = "platforms"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    short_name: Mapped[str] = mapped_column(String(20), nullable=False, index=True)
    type: Mapped[PlatformType] = mapped_column(
        Enum(PlatformType, name="platform_type"), nullable=False
    )
    is_system: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

    platform_accounts = relationship("PlatformAccount", back_populates="platform")
